### PR TITLE
fix(#3): return error in pre-commit on new output directories

### LINKED ISSUE #3: No failure when used as pre-commit hook
**Describe the bug**

I used this as a pre-commit hook. The hook generated new files. Git commit passed OK, instead of failing.

#**To Reproduce**

Steps and/or code snippet(s) to reproduce the behavior:

1. Add a new output.
1. Configure the pre-commit hook.
2. Commit.

**Expected behavior**

It should fail because it created new files. That means the template output is out of sync, and thus there's a hook failure.

**Additional Information**

- `copier_template_tester` Version:  1.0.2
- OS: \[e.g. macOS/Windows\]: Linux
- OS Version \[e.g. Catalina/10\]: Fedora 37

### PR DESCRIPTION
**Pull request by KyleKing**
Fix #3

Using [`git ls-file` and a few options](https://stackoverflow.com/a/3801554/3219667), I can get the list of untracked directories. This should be cross-platform, but needs unit tests and to ensure edge cases are handled and that it works within pre-commit

### CODE REVIEW

File: copier_template_tester/main.py [hunk 1]
```diff
@@ -63,11 +82,14 @@ def run(base_dir: Path | None = None) -> None:
     defaults = config.get('defaults', {})
 
     input_path = base_dir
+    output_paths = set()
     for key, data in config['output'].items():
         output_path = base_dir / key
         output_path.mkdir(parents=True, exist_ok=True)
+        output_path.add(output_path)
```
Comment (yajo) [inline copier_template_tester/main.py:89]: I tested it and it failed with:

```

copier-template-tester...................................................Failed
- hook id: copier-template-tester
- exit code: 1

Warning: You probably want a section: [defaults]
Traceback (most recent call last):
  File "/nix/store/3dq3n73wf7zc5j51bc7sr3939mbvdxp4-python3.10-copier-template-tester-1.0.2/bin/.ctt-wrapped", line 9, in <module>
    sys.exit(run())
  File "<@beartype(copier_template_tester.main.run) at 0x7fd9ecd50280>", line 36, in run
  File "/nix/store/a0dqd5bia22xinhwxp241c59376xf5ld-python3-3.10.9-env/lib/python3.10/site-packages/copier_template_tester/main.py", line 89, in run
    output_path.add(output_path)
AttributeError: 'PosixPath' object has no attribute 'add'
```

**Review by yajo — CHANGES_REQUESTED** [review]

File: copier_template_tester/main.py (hunk 1 above)
Comment (KyleKing) [reply copier_template_tester/main.py]: Thanks for bumping this issue and trying out the changes!

**Comment by KyleKing** [comment]
Merging so that I can open a new PR to improve the test suite
